### PR TITLE
Pool create/update with invalid cluster-issuer should report error

### PR DESCRIPTION
**Pull request by aleksej-paschenko**
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: cmd/ketch/cluster_issuer.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var (
```
Comment (itsksaurabh) [inline cmd/ketch/cluster_issuer.go:12]: could we avoid global vars? how about `getclusterIssuerGVR()` ?

**Review by itsksaurabh — CHANGES_REQUESTED** [review]

File: cmd/ketch/configuration/configuration.go [hunk 2]
```diff
@@ -70,3 +72,18 @@ func (cfg *Configuration) Storage() templates.Client {
 	cfg.storage = templates.NewStorage(cfg.Client(), controllers.KetchNamespace)
 	return cfg.storage
 }
+
+// DynamicClient returns kubernetes dynamic client. It's used to work with CRDs for which we don't have go types like ClusterIssuer.
```
[the remark this answers is not in the document]
Comment (aleksej-paschenko) [reply cmd/ketch/configuration/configuration.go:76]: not sure that I understand, what do you want to import?

File: cmd/ketch/cluster_issuer.go (hunk 1 above)
Comment (aleksej-paschenko) [reply cmd/ketch/cluster_issuer.go:12]: what is the problem with this type of usage?

File: cmd/ketch/configuration/configuration.go [hunk 3]
```diff
@@ -70,3 +72,18 @@ func (cfg *Configuration) Storage() templates.Client {
...
+func (cfg *Configuration) DynamicClient() dynamic.Interface {
+	flags := genericclioptions.NewConfigFlags(true)
+	factory := cmdutil.NewFactory(flags)
+	conf, err := factory.ToRESTConfig()
+	if err != nil {
+		log.Fatalf("can't create kubernetes client: %v", err)
```
[the remark this answers is not in the document]
Comment (aleksej-paschenko) [reply cmd/ketch/configuration/configuration.go:82]: fixed

File: cmd/ketch/cluster_issuer.go (hunk 1 above)
Comment (aleksej-paschenko) [reply cmd/ketch/cluster_issuer.go]: remove the var declaration

File: cmd/ketch/cluster_issuer.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	clusterIssuerGVR = schema.GroupVersionResource{
+		Group:    "cert-manager.io",
+		Version:  "v1",
+		Resource: "clusterissuers",
+	}
+)
+
+func isClusterIssuerExist(iface dynamic.Interface, ctx context.Context, clusterIssuerName string) (bool, error) {
```
[the remark this answers is not in the document]
Comment (aleksej-paschenko) [reply cmd/ketch/cluster_issuer.go]: renamed but I don't think we need nil checks now

File: cmd/ketch/configuration/configuration.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (itsksaurabh) [reply cmd/ketch/configuration/configuration.go:76]: import?

**Review by itsksaurabh — APPROVED** [review]

**Pull request merged**